### ISSUE TITLE
Header preview when opening FITS file displays * character as . character

### ISSUE THREAD

**Issue by victorliu1231** [CARTAvis/carta-backend#1423]
**Describe the bug**
When opening a FITS file with file browser UI popup, the "File Information" and "Header" previews display the * character as a . character in the BUNIT keyword. E.g. if 'BUNIT = K (Ta*)' in the header, it shows up as 'BUNIT = K. (Ta.)' and 'Pixel unit = K. (Ta.)' in the frontend.

**To Reproduce**
1. Go to 'File' --> 'Open Image'
2. Click on any FITS file where the BUNIT keyword has a * character.
3. Scroll down to 'Pixel unit' in 'File Information' and 'BUNIT =' in 'Header'
4. See error.

**Expected behavior**
The frontend should display the * character properly. E.g. if 'BUNIT = K (Ta*)' in the header, it shows up as 'BUNIT = K (Ta*)' and 'Pixel unit = K (Ta*)' in the frontend.

**Screenshots or videos**
![frontend_file_information](https://github.com/user-attachments/assets/6a6e698f-15d9-4a77-964a-377bc913b242)
![frontend_header](https://github.com/user-attachments/assets/10f06a56-20fd-4f75-847d-545c98ce7c63)
![command_line_header](https://github.com/user-attachments/assets/23dd6247-c7b3-4f06-87d5-5174118f8e07)

**Platform info (please complete the following information):**
 - OS: Linux RHEL
 - Browser: Mozilla Firefox
 - Browser version: 128.4.0esr (64-bit)
 - CARTA version: 4.1.0 (18 Jan 2024)

**Comment by kswang1029** [comment]
@victorliu1231 could you provide a test FITS file for investigation please?

**Comment by victorliu1231** [comment]
Yes, here is a test file. It appears GitHub does not let me upload FITS files, so I have provided a Google Drive link.
https://drive.google.com/file/d/1LlCeqbQ1_2MSkgzO6dyXN2f0lmPiZ-3e/view?usp=sharing

**Comment by loveluthien** [comment]
@victorliu1231 Thanks for sharing, we'll deal with that.

**Comment by loveluthien** [comment]
@kswang1029 @YuHsuan-Hwang it seems to be a backend or casa issue.

**Comment by kswang1029** [comment]
@confluence could you arrange this fix for v5-stable please?

**Comment by confluence** [comment]
@pford I have confirmed that the incorrect value is coming from the backend.

**Comment by veggiesaurus** [comment]
I'm fairly certain this is a casacore issue, because setting `BUNIT` to `Hello*World` works fine, as does `A*B`, but `K*T` gets turned into `K.T`, presumably because it parses those as units

**Comment by pford** [comment]
@kswang1029 I have been looking into how the backend reads the headers from FITS, but then "normalizes" the units to a casacore::Unit, resulting in the issue noted here.  For some units, this is so that the quantities can be converted, such as reporting the reference coordinates in degrees and restoring beam in arcsec and degrees.  But changing the unit could be done for the conversion only. Do you think the file list File Information and Headers should contain the original header units for all images? Although not the original value, the new unit is sometimes more readable to the user (`beam-1 Jy` -> `Jy/beam`).

**Comment by kswang1029** [comment]
I wonder if we can consider Ta* (and TA*) a special case and handle it separately （meaning we don't change the overall behavior as is now)? The current approach seems working fine mostly based on user feedback (almost none?).

I will check some other images and compare the unit parsing part before and after to see if there are potential issues I missed.

**Comment by pford** [comment]
OK, I just recall questions about "My header is X, why does it show Y?  Other software shows header X"

I have fixed this issue so the unit string in GILDAS format is unchanged.  Flux density still works since it starts with "K".  Will open a pull request.

**Comment by pford** [comment]
Other software has had the same issue, such as radio-astro-tools / spectral-cube (https://github.com/radio-astro-tools/spectral-cube/issues/213) which uses astropy.units.  They suggest changing the unit from "K (Tmb)" to "K"!

Since there can be different temperature types in the unit, any unit with format "K (Tx)", where x is alpha and *, will be treated as a GILDAS unit and protected from casacore in the file list and frontend.  casacore::FITSImage internally changes its unit to "K.(Ta.)" but this will be invisible to the user.

**Comment by kswang1029** [comment]
Ok I think this solution is good. Will do some tests.
